index.d.ts

```diff
@@ -1,5 +1,7 @@
 import {Primitive, JsonObject} from 'type-fest';
 
+export {default as errorConstructors} from './error-constructors.js';
```

Comment:
This adds support for custom error constructors to be added. This only supports simple errors so I mentioned this in the docs and in the types. A new issue can be created to add support for such error constructors.

Also, instead of adding `error-constructors.js` to the exports map, I just added another export here. It's cleaner anyway.